src/tactic/transform_decl.lean

```diff
@@ -44,30 +44,43 @@ meta def additive_test (f : name → option name) (replace_all : bool) (ignore :
   (e : expr) : bool :=
 if replace_all then tt else additive_test_aux f ignore ff e
 
-private meta def transform_decl_with_prefix_fun_aux (f : name → option name)
+/-- transform the declaration `src` and all declarations `pre._proof_i` occurring in `src`
+using the dictionary `f`.
+`replace_all`, `trace`, `ignore` and `reorder` are configuration options.
+`pre` is the declaration that got the `@[to_additive]` attribute and `tgt_pre` is the target of this
+declaration. -/
+meta def transform_decl_with_prefix_fun_aux (f : name → option name)
   (replace_all trace : bool) (ignore reorder : name_map $ list ℕ) (pre tgt_pre : name)
   (attrs : list name) : name → command :=
 λ src,
 do
+  -- if this declaration is not `pre` or an internal declaration, we do nothing.
+  tt ← return (src = pre ∨ src.is_internal : bool) |
```

Comment:
Isn't this just
```suggestion
  guard (src = pre ∨ src.is_internal) |
```
?

Reply:
No, `|` syntax only applies for pattern matching.

Reply:
More at [the tactic writing tutorial](https://leanprover-community.github.io/extras/tactic_writing.html):

> Lean also provides a dedicated syntax for pattern matching with a single pattern and a wildcard, where execution only proceeds to the next line of the do block if the pattern matches, and otherwise executes the expression after the |:
> ```lean
> meta def trace_goal_is_eq : tactic unit :=
> do `(%%l = %%r) ← tactic.target | tactic.trace "Goal is not an equality",
>    tactic.trace $ "Goal is equality between " ++ (to_string l) ++ " and " ++ (to_string r)
> ```

Reply:
As Eric said, this way of coding ensures that the rest of the code is skipped if the pattern doesn't apply.
It's hard to mimic with `<|>` in this case, since there is also a `fail` after the pipe `|`.
It is equivalent to `if src = pre ∨ src.is_internal then ... else ...`, but I like this better, since it avoids extra indentation for the rest of the do block.

Reply:
Ah, I didn't realize the `|` operator is only for matching. Sorry!